CWorld.cpp

```diff
@@ -170,27 +169,107 @@ void CWorld::InitWorldFromMap(const Map& map) {
 void CWorld::CollideEntitiesWithWorld() {
     std::vector<BaseGameEntity*> entities = EntityManager::Instance()->Entities();
     double dt = GetDeltaTime();
-    for (int i = 0; i < entities.size(); i++) {
-        BaseGameEntity* pEntity = entities[i];
+    static double accumulator = 0.0;
+    accumulator += dt;
+    int numUpdateSteps = 0;
 
-        // FIX: Higher level entity type or entity flags (eg. FLAG_COLLIDABLE).
-        if ( (pEntity->Type() == ET_PLAYER) || (pEntity->Type() == ET_ENEMY) ) {
+    while (accumulator >= DOD_FIXED_UPDATE_TIME) {
```

Comment:
Sure.

The fixed update rate is defined as:
```cpp
DOD_FIXED_UPDATE_TIME = 1000.0/60.0;
```

which is 16.67ms. So the accumulator adds milliseconds every frame until it is larger than 16.67ms => 60Hz. Then, as long as the accumulator is larger than that the movement/collision logic gets updated. Think of it as **integration** (in math terms) over time: The magnitude of **positional change** is integrated over time, not caring about the framerate. So the magnitude is always the same because we always integrate over the same boundaries, thus independet of the framerate.
The same approach is used in the movement prediction code (in `g_fp_player.cpp`. The only pitfall there is that keyboard inputs *must* be taken every frame and then caching them for the next movement update. The velocity, of course, is still updated every frame. At the moment there is no lerp done for velocity as I didn't see the need of it in my tests. Unlike in the collision detection and response in `CWorld`: There we cache the previous position so we can linearly interpolate between the current position (which only gets updated every 16.67ms) and the last time the position was updated. This is very important as otherwise you would see stuttering (looks like micro-stuttering when the GPU is CPU bound and its pipeline is being stalled for a few ms).